### PR TITLE
Confirm

### PR DESCRIPTION
**Pull request by INR4GE**

### CODE REVIEW

File: utils/sendmail.js [hunk 1]
```diff
@@ -24,4 +23,47 @@ const passwordResetEmail = (text, firstName) => `
...
+    line-height: 2;
+    font-size: 1.1rem;
+  ">
+    <h2>Hello, ${firstName}</h2>
+    <p>Please confirm email</p>
+    <a href=${process.env.FRONT_BASE_URI}confirmation/${token}>Confirm email</a> 
+  </div>
+`;
+
+const sendEmail = async (message, callback) => {
```
Comment (Alan18081) [inline utils/sendmail.js:40]: Please, refactor it to use promise. Here you have an argument `callback` but you don't pass it on `sendEmail` call. You need to use `new Promise((resolve, reject) => {...})` to properly map callback-based function to promise-based

File: modules/user/user.service.js [hunk 2]
```diff
@@ -22,16 +30,18 @@ class UserService {
     }
   }
 
-  async getUserByFieldOrThrow(key, param) {
+  async getUserByFieldOrThrow(key, param, customError) {
```
Comment (Alan18081) [inline modules/user/user.service.js:33]: Why do we need `customError`?

File: modules/user/user.service.js [hunk 3]
```diff
@@ -101,17 +107,24 @@ class UserService {
...
+    );
 
     const match = await bcrypt.compare(
       password,
       user.credentials.find(cred => cred.source === 'horondi').tokenPass,
     );
 
     if (!match) {
-      throw new AuthenticationError('Wrong password');
+      throw new AuthenticationError(WRONG_CREDENTIALS[language].value);
```
Comment (Alan18081) [inline modules/user/user.service.js:122]: Revert this change, please, we are going to send back unlocalized error messages and localize them only on backend

File: modules/user/user.service.js [hunk 4]
```diff
@@ -101,17 +107,24 @@ class UserService {
       throw new UserInputError('Errors', { errors });
     }
 
-    const user = await this.getUserByFieldOrThrow('email', email);
+    const user = await this.getUserByFieldOrThrow(
+      'email',
+      email,
+      WRONG_CREDENTIALS[language].value,
```
Comment (Alan18081) [inline modules/user/user.service.js:113]: Revert this change, please, we are going to send back unlocalized error messages and localize them only on backend

File: modules/user/user.service.js [hunk 5]
```diff
@@ -101,17 +107,24 @@ class UserService {
...
       user.credentials.find(cred => cred.source === 'horondi').tokenPass,
     );
 
     if (!match) {
-      throw new AuthenticationError('Wrong password');
+      throw new AuthenticationError(WRONG_CREDENTIALS[language].value);
     }
 
+    if (!user.confirmed) {
+      return new Error(NOT_CONFIRMED[language].value);
```
Comment (Alan18081) [inline modules/user/user.service.js:126]: Revert this change, please, we are going to send back unlocalized error messages and localize them only on backend

File: modules/user/user.service.js [hunk 6]
```diff
@@ -151,11 +164,34 @@ class UserService {
...
   deleteUser(id) {
     return User.findByIdAndDelete(id);
   }
+
+  async confirmUser(token, language) {
+    try {
+      const decoded = jwt.verify(token, process.env.SECRET);
+      const user = await User.findOne({ email: decoded.email });
+      if (!user) {
+        return new Error(TOKEN_NOT_VALID[language].value);
```
Comment (Alan18081) [inline modules/user/user.service.js:187]: Revert this change, please, we are going to send back unlocalized error messages and localize them only on backend

File: modules/user/user.service.js [hunk 7]
```diff
@@ -151,11 +164,34 @@ class UserService {
...
+      }
+      user.confirmed = true;
+      await User.findByIdAndUpdate(user._id, user);
+      return true;
+    } catch (e) {
+      return new Error(CONFIRMATION_ERROR[language].value);
```
Comment (Alan18081) [inline modules/user/user.service.js:193]: Revert this change, please, we are going to send back unlocalized error messages and localize them only on backend

File: modules/user/user.service.js [hunk 8]
```diff
@@ -151,11 +164,34 @@ class UserService {
       ],
     });
     const savedUser = await user.save();
+    const token = await generateToken(savedUser._id, savedUser.email);
+    const message = {
+      from: `horondi.devproject@gmail.com`,
```
Comment (Alan18081) [inline modules/user/user.service.js:169]: It should be taken from the environment variable

File: modules/user/user.resolver.js [hunk 9]
```diff
@@ -7,14 +7,15 @@ const userQuery = {
 };
 const userMutation = {
   registerUser: (parent, args) => userService.registerUser(args.user),
-  loginUser: (parent, args) => userService.loginUser(args.user),
+  loginUser: (parent, args) => userService.loginUser(args.user, args.language),
```
Comment (Alan18081) [inline modules/user/user.resolver.js:10]: Revert this change, please, we are going to send back unlocalized error messages and localize them only on backend

File: modules/user/user.resolver.js [hunk 10]
```diff
@@ -7,14 +7,15 @@ const userQuery = {
...
   deleteUser: (parent, args) => userService.deleteUser(args.id),
   updateUserById: (parent, args, context) => (context.user
     ? userService.updateUserById(args.user, args.id)
     : new Error('Unauthorized')),
   updateUserByToken: (parent, args, context) => (context.user
-    ?  userService.updateUserByToken(args.user, context.user)
+    ? userService.updateUserByToken(args.user, context.user)
     : new Error('Unauthorized')),
+  confirmUser: (parent, args) => userService.confirmUser(args.token, args.language),
```
Comment (Alan18081) [inline modules/user/user.resolver.js:18]: Revert this change, please, we are going to send back unlocalized error messages and localize them only on backend

File: typeDefs.js [hunk 11]
```diff
@@ -197,10 +197,11 @@ const typeDefs = gql`
 
     "User Mutation"
     registerUser(user: UserInput!): User
-    loginUser(user: UserInput!): User
+    loginUser(user: UserInput!, language: Int!): User
```
Comment (Alan18081) [inline typeDefs.js:200]: Revert this change, please, we are going to send back unlocalized error messages and localize them only on backend

File: typeDefs.js [hunk 12]
```diff
@@ -197,10 +197,11 @@ const typeDefs = gql`
...
     deleteUser(id: ID!): User
     updateUserById(user: UserInput!, id: ID!): User
     updateUserByToken(user: UserInput!): User
+    confirmUser(token: String!, language: Int!): Boolean
```
Comment (Alan18081) [inline typeDefs.js:204]: Revert this change, please, we are going to send back unlocalized error messages and localize them only on backend

File: error-messages/user.messages.js [hunk 13]
```diff
@@ -0,0 +1,42 @@
+const CONFIRMATION_ERROR = [
```
Comment (Alan18081) [inline error-messages/user.messages.js:1]: Please. remove these translations because now we don't return translations from backend anymore. We just return uppercase error message. Also, please, make error messages a little bit more informative, provide a clear reason, why the error has happened. `Oops! Something went wrong!` is not informative for the client

File: utils/sendmail.js (hunk 1 above)
Comment (INR4GE) [reply utils/sendmail.js]: Refactored

File: modules/user/user.service.js (hunk 2 above)
Comment (INR4GE) [reply modules/user/user.service.js]: Removed

File: modules/user/user.service.js (hunk 3 above)
Comment (INR4GE) [reply modules/user/user.service.js]: Fixed

File: modules/user/user.service.js (hunk 5 above)
Comment (INR4GE) [reply modules/user/user.service.js]: removed

File: modules/user/user.service.js (hunk 4 above)
Comment (INR4GE) [reply modules/user/user.service.js]: removed

File: modules/user/user.service.js (hunk 6 above)
Comment (INR4GE) [reply modules/user/user.service.js]: removed

File: modules/user/user.service.js (hunk 7 above)
Comment (INR4GE) [reply modules/user/user.service.js]: removed

File: modules/user/user.service.js (hunk 8 above)
Comment (INR4GE) [reply modules/user/user.service.js]: fixed

File: modules/user/user.resolver.js (hunk 9 above)
Comment (INR4GE) [reply modules/user/user.resolver.js]: removed

File: modules/user/user.resolver.js (hunk 10 above)
Comment (INR4GE) [reply modules/user/user.resolver.js]: removed

File: typeDefs.js (hunk 11 above)
Comment (INR4GE) [reply typeDefs.js]: removed

File: typeDefs.js (hunk 12 above)
Comment (INR4GE) [reply typeDefs.js]: removed

File: utils/createToken.js [hunk 14]
```diff
@@ -1,9 +1,12 @@
 const jwt = require('jsonwebtoken');
```
Comment (Alan18081) [inline utils/createToken.js:1]: Please, rename this file to the `create-token` to follow a common naming patterns

Comment (INR4GE) [reply utils/createToken.js]: renamed
